wpilibc/src/main/native/cpp/Joystick.cpp

```diff
@@ -119,5 +119,8 @@ double Joystick::GetMagnitude() const {
 }
 
 units::radian_t Joystick::GetDirection() const {
+  // https://docs.wpilib.org/en/stable/docs/software/basic-programming/coordinate-system.html#joystick-and-controller-coordinate-system
+  // +X is right and +Y is back, so 0 radians is right and CW is positive.
```

Comment:
You're conflating translation with rotation around axes. +X is forward and +Y is right as per the NED axes convention. Pushing the joystick forward is negative rotation around the y-axis, and pushing the joystick to the right is positive rotation around the x-axis.